diff --git a/src/click/core.py b/src/click/core.py
index 6dc44f3..f0a624b 100644
--- a/src/click/core.py
+++ b/src/click/core.py
@@ -3110,7 +3110,7 @@ class Option(Parameter):
                 )[1]
             elif self.is_bool_flag and not self.secondary_opts and not default_value:
                 default_string = ""
-            elif isinstance(default_value, str) and default_value == "":
+            elif default_value == "":
                 default_string = '""'
             else:
                 default_string = str(default_value)
