diff --git a/src/jinja2/utils.py b/src/jinja2/utils.py
index 7b52fc03..5c1ff5d7 100644
--- a/src/jinja2/utils.py
+++ b/src/jinja2/utils.py
@@ -18,17 +18,8 @@ if t.TYPE_CHECKING:
 
 F = t.TypeVar("F", bound=t.Callable[..., t.Any])
 
-
-class _MissingType:
-    def __repr__(self) -> str:
-        return "missing"
-
-    def __reduce__(self) -> str:
-        return "missing"
-
-
-missing: t.Any = _MissingType()
-"""Special singleton representing missing values for the runtime."""
+# special singleton representing missing values for the runtime
+missing: t.Any = type("MissingType", (), {"__repr__": lambda x: "missing"})()
 
 internal_code: t.MutableSet[CodeType] = set()
 
