diff --git a/src/jinja2/compiler.py b/src/jinja2/compiler.py
index 23295ec1..074e9b18 100644
--- a/src/jinja2/compiler.py
+++ b/src/jinja2/compiler.py
@@ -1141,14 +1141,9 @@ class CodeGenerator(NodeVisitor):
             )
             self.writeline(f"if {frame.symbols.ref(alias)} is missing:")
             self.indent()
-            # The position will contain the template name, and will be formatted
-            # into a string that will be compiled into an f-string. Curly braces
-            # in the name must be replaced with escapes so that they will not be
-            # executed as part of the f-string.
-            position = self.position(node).replace("{", "{{").replace("}", "}}")
             message = (
                 "the template {included_template.__name__!r}"
-                f" (imported on {position})"
+                f" (imported on {self.position(node)})"
                 f" does not export the requested name {name!r}"
             )
             self.writeline(
