diff --git a/lib/command.js b/lib/command.js
index 9484a5e..93b9d01 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1040,15 +1040,15 @@ Expecting one of '${allowedValues.join("', '")}'`);
     }
 
     // By default terminate process when spawned process terminates.
+    // Suppressing the exit if exitCallback defined is a bit messy and of limited use, but does allow process to stay running!
     const exitCallback = this._exitCallback;
-    proc.on('close', (code, _signal) => {
-      code = code ?? 1; // code is null if spawned process terminated due to a signal
-      if (!exitCallback) {
-        process.exit(code);
-      } else {
-        exitCallback(new CommanderError(code, 'commander.executeSubCommandAsync', '(close)'));
-      }
-    });
+    if (!exitCallback) {
+      proc.on('close', process.exit.bind(process));
+    } else {
+      proc.on('close', () => {
+        exitCallback(new CommanderError(process.exitCode || 0, 'commander.executeSubCommandAsync', '(close)'));
+      });
+    }
     proc.on('error', (err) => {
       // @ts-ignore
       if (err.code === 'ENOENT') {
