diff --git a/lib/option.js b/lib/option.js
index d61fc5f..d8a9a15 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -99,12 +99,7 @@ class Option {
    * @return {Option}
    */
   implies(impliedOptionValues) {
-    let newImplied = impliedOptionValues;
-    if (typeof impliedOptionValues === 'string') {
-      // string is not documented, but easy mistake and we can do what user probably intended.
-      newImplied = { [impliedOptionValues]: true };
-    }
-    this.implied = Object.assign(this.implied || {}, newImplied);
+    this.implied = Object.assign(this.implied || {}, impliedOptionValues);
     return this;
   }
 
