diff --git a/lib/router/index.js b/lib/router/index.js
index fbe94acd..69e6d380 100644
--- a/lib/router/index.js
+++ b/lib/router/index.js
@@ -287,12 +287,6 @@ proto.handle = function handle(req, res, out) {
 
   function trim_prefix(layer, layerError, layerPath, path) {
     if (layerPath.length !== 0) {
-      // Validate path is a prefix match
-      if (layerPath !== path.substr(0, layerPath.length)) {
-        next(layerError)
-        return
-      }
-
       // Validate path breaks on a path separator
       var c = path[layerPath.length]
       if (c && c !== '/' && c !== '.') return next(layerError)
