diff --git a/src/markupsafe/_speedups.c b/src/markupsafe/_speedups.c
index 09dd57c..73c2955 100644
--- a/src/markupsafe/_speedups.c
+++ b/src/markupsafe/_speedups.c
@@ -151,7 +151,7 @@ escape_unicode_kind4(PyUnicodeObject *in)
 static PyObject*
 escape_unicode(PyObject *self, PyObject *s)
 {
-	if (!PyUnicode_Check(s))
+	if (!PyUnicode_CheckExact(s))
 		return NULL;
 
     // This check is no longer needed in Python 3.12.
